iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl

```diff
@@ -26,19 +26,28 @@ namespace iox
 {
 namespace cxx
 {
+/// m_data fields are explicitly initialized whenever a new element is inserted
+/// into the list
+/// @NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init,hicpp-member-init)
```

Comment:
I'm not sure what happens when there is a doxygen comment without a doxygen tag. I would suggest to always use `@NOLINTJUSTIFICATION` with doxygen comments